src/signing.rs

```diff
@@ -707,9 +712,15 @@ impl From<&SecretKey> for ExpandedSecretKey {
         // TODO: Use bytes.split_array_ref once it’s in MSRV.
         let (lower, upper) = hash.split_at(32);
 
+        // Clamp the lower bytes. This is the key integer.
         // The try_into here converts to fixed-size array
+        let key_bytes = clamp_integer(lower.try_into().unwrap());
+        // For signing, we'll need the integer as a Scalar
+        let key = Scalar::from_bytes_mod_order(key_bytes);
```

Comment:
Previously this was unreduced. We need to ensure that this is used in a way that is consistent with how the unreduced scalar worked. Indeed this is the case. The only computation that `self.key` is used in is in `ExpandedSecretKey::{sign, sign_prehashed}`, where it's multiplied by some scalar. `Scalar::mul` is defined to work the same whether or not the inputs are reduced, so the computation is unchanged. I added the `test_mul_reduction_invariance` unit test to `curve25519-dalek` to make sure of this.